test/unit/notify/send-email.test.js

```diff
@@ -1,164 +1,250 @@
-  })
-
-  test('should call notifyClient.sendEmail when a valid message is received', async () => {
-    await sendEmail(mockContext, mockMessage)
-
-    const notifyClientMockInstance = notifyClient.mock.instances[0]
-    expect(notifyClientMockInstance.sendEmail).toHaveBeenCalledTimes(1)
-  })
-
-  test('should not call notifyClient.sendEmail with empty reference', async () => {
-    await sendEmail(mockContext, mockMessage)
-
-    const notifyClientMockInstance = notifyClient.mock.instances[0]
-    expect(notifyClientMockInstance.sendEmail).not.toHaveBeenCalledWith(env.notifyEmailTemplateId, env.notifyEmailAddress, {
-      personalisation: flatten(mockMessage),
-      reference: ''
-    })
-  })
-
-  test('should  call notifyClient.sendEmail with uuid when no reference is provided', async () => {
-    await sendEmail(mockContext, mockMessage)
-
-    const notifyClientMockInstance = notifyClient.mock.instances[0]
-    expect(notifyClientMockInstance.sendEmail).toHaveBeenCalledWith(env.notifyEmailTemplateId, env.notifyEmailAddress, {
-      personalisation: flatten(mockMessage),
-      reference: mockReference
-    })
-  })
-
-  test('should call notifyClient.sendEmail with correct parameters when a emailAddress and reference are received', async () => {
-    await sendEmail(mockContext, mockMessage, emailAddress, mockReference)
-
-    const notifyClientMockInstance = notifyClient.mock.instances[0]
-    expect(notifyClientMockInstance.sendEmail).toHaveBeenCalledWith(env.notifyEmailTemplateId, emailAddress, {
-      personalisation: flatten(mockMessage),
-      reference: mockReference
-    })
-  })
-
-  test('should throw error when sendEmail rejects', async () => {
-    notifyClient.prototype.sendEmail.mockRejectedValue({ Error: 'Request failed with status code 403' })
-
-    const wrapper = async () => {
-      await sendEmail(mockContext, mockMessage)
-    }
-
-    expect(wrapper).rejects.toThrow()
-  })
-
-  test('should throw Error when sendEmail rejects', async () => {
-    notifyClient.prototype.sendEmail.mockRejectedValue({ Error: 'Request failed with status code 403' })
-
-    const wrapper = async () => {
-      await sendEmail(mockContext, mockMessage)
-    }
-
-    expect(wrapper).rejects.toThrow(Error)
-  })
-
-  test('should throw an error which starts with "Oh dear" when sendEmail rejects', async () => {
-    notifyClient.prototype.sendEmail.mockRejectedValue({ Error: 'Request failed with status code 403' })
-
-    const wrapper = async () => {
-      await sendEmail(mockContext, mockMessage)
-    }
-
-    expect(wrapper).rejects.toThrowError(/^Oh dear/)
-  })
-})
+const flatten = require('flat')
+const env = require('../../env')
+const mockContext = require('../../mock-context')
+const mockMessage = require('../../mock-context')
+const mockReference = require('../../mock-reference')
+
+let notifyClient
+let validateEmail
+let sendEmail
+let emailAddress
+let defaultEmailAddress
+let message
+let eventTemplates

```

Comment:
could we import the actual `eventTemplates` config?

or could we mock these similar to some of the mocks above?